dvc/cache/base.py

```diff
@@ -537,17 +537,19 @@ def checkout(
         progress_callback=None,
         relink=False,
         filter_info=None,
+        quiet=False,
```

Comment:
Stage/output `checkout()` take `allow_missing` parameter now for checkpoints, however there may be situations where we want to allow missing outputs but still output the warnings (like for `dvc checkout` command - #4746).

So at the cache level, we just use `quiet` to suppress warnings, but still return the failed checkout info to be handled or allowed at the output/stage level.